src/nunit-gui/Model/TestModel.cs

```diff
@@ -350,39 +348,33 @@ public void OnTestEvent(string report)
             switch (xmlNode.Name)
             {
                 case "start-test":
-                    if (TestStarting != null)
-                        TestStarting(new TestNodeEventArgs(TestAction.TestStarting, new TestNode(xmlNode)));
+                    TestStarting?.Invoke(new TestNodeEventArgs(TestAction.TestStarting, new TestNode(xmlNode)));
                     break;
 
                 case "start-suite":
-                    if (SuiteStarting != null)
-                        SuiteStarting(new TestNodeEventArgs(TestAction.SuiteStarting, new TestNode(xmlNode)));
+                    SuiteStarting?.Invoke(new TestNodeEventArgs(TestAction.SuiteStarting, new TestNode(xmlNode)));
                     break;
 
                 case "start-run":
-                    if (RunStarting != null)
-                        RunStarting(new RunStartingEventArgs(xmlNode.GetAttribute("count", -1)));
+                    RunStarting?.Invoke(new RunStartingEventArgs(xmlNode.GetAttribute("count", -1)));
                     break;
 
                 case "test-case":
                     ResultNode result = new ResultNode(xmlNode);
                     _resultIndex[result.Id] = result;
-                    if (TestFinished != null)
-                        TestFinished(new TestResultEventArgs(TestAction.TestFinished, result));
+                    TestFinished?.Invoke(new TestResultEventArgs(TestAction.TestFinished, result));
                     break;
 
                 case "test-suite":
                     result = new ResultNode(xmlNode);
                     _resultIndex[result.Id] = result;
-                    if (SuiteFinished != null)
-                        SuiteFinished(new TestResultEventArgs(TestAction.TestFinished, result));
+                    SuiteFinished?.Invoke(new TestResultEventArgs(TestAction.TestFinished, result));
```

Comment:
All the translations to conditional member access looks correct (and also we are now threadsafe ;) ). The only problem I discovered was that we are using the wrong `TestAction`. It should be `TestAction.SuiteFinished` not `TestAction.TestFinished`.